klayout/drc/testing/run_regression.py

```diff
@@ -0,0 +1,924 @@
+# Copyright 2022 GlobalFoundries PDK Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Run GlobalFoundries 180nm MCU DRC Unit Regression.
+
+Usage:
+    run_regression.py (--help| -h)
+    run_regression.py [--mp=<num>] [--run_name=<run_name>] [--table_name=<table_name>]
+
+Options:
+    --help -h                           Print this help message.
+    --mp=<num>                          The number of threads used in run.
+    --run_name=<run_name>               Select your run name.
+    --table_name=<table_name>           Target specific table.
+"""
+
+from subprocess import check_call
+from subprocess import Popen, PIPE
+import concurrent.futures
+import traceback
+import yaml
+from docopt import docopt
+import os
+from datetime import datetime
+import xml.etree.ElementTree as ET
+import time
+import pandas as pd
+import logging
+import glob
+from pathlib import Path
+from tqdm import tqdm
+import re
+import gdstk
+import errno
+
+from collections import defaultdict
+
+
+SUPPORTED_TC_EXT = "gds"
+SUPPORTED_SW_EXT = "yaml"
+RULE_LAY_NUM = 10000
+PATH_WIDTH = 0.01
+RULE_STR_SEP = "--"
+ANALYSIS_RULES = ["pass_patterns", "fail_patterns", "false_negative", "false_positive"]
+
+
+def get_unit_test_coverage(gds_file):
+    """
+    This function is used for getting all test cases available inside a single test table.
+    Parameters
+    ----------
+    gds_file : str
+        Path string to the location of unit test cases path.
+    Returns
+    -------
+    list
+        A list of unique rules found.
+    """
+    # Get rules from gds
+    rules = []
+
+    # layer num of rule text
+    lay_num = 11
+    # layer data type of rule text
+    lay_dt = 222
+
+    # Getting all rules names from testcase
+    library = gdstk.read_gds(gds_file)
+    top_cells = library.top_level()  # Get top cells
+    for cell in top_cells:
+        flatten_cell = cell.flatten()
+        # Get all text labels for each cell
+        labels = flatten_cell.get_labels(apply_repetitions=True, depth=None, layer=lay_num, texttype=lay_dt)
+        # Get label value
+        for label in labels:
+            rule = label.text
+            if rule not in rules:
+                rules.append(rule)
+
+    return rules
+
+
+def check_klayout_version():
```

Comment:
should that go in a share module (since we do the same in `run_drc`)

Reply:
We could do that. No problem. But I have a few questions, do you want to just have an extra file or build a module dir

Reply:
maybe it could be imported from `run_drc` itself?

Reply:
I'll create a `util.py` file to have all such procedures. What do you think?

Reply:
I think loading the function from `run_drc` is totally acceptable here, and avoid the current duplication.

Reply:
#6